### PR TITLE
Adding the rare gun H&K USC 45 carbine

### PR DESCRIPTION
**Pull request by Ilurker**
#### Summary
Content "Adds rare gun - H&K USC 45 Carbine"

#### Purpose of change

Adds the uncommon, civilian carbine variant of the UMP submachine gun, the USC 45 Carbine. This fills a neglected role for 45 ACP ammunition, which does not have any rifles in its niche, while providing the interesting addition of a high-accurary, low damage long gun.

#### Describe the solution

Adds the USC 45 carbine and its magazines to the rare gun spawn tables. All stats were pulled from manufacturer pages whenever possible and damage was assigned per the table under GAME_BALANCE.md

#### Describe alternatives you've considered

Feeling kind of sad whenever I loot yet another UMP I'll never use.

#### Testing

Spawned all items and verified they behaved as expected.

#### Additional context

https://hk-usa.com/hk-models/usc/
https://promagindustries.com/hk-usc-45-acp-carbine-10-rd-black-polymer/
https://promagindustries.com/hk-usc-45-acp-carbine-15-rd-black-polymer/
https://promagindustries.com/hk-usc-45-acp-carbine-20-rd-black-polymer/
https://hk-usa.com/wp-content/uploads/USC-Operators-Manual-8_3_20.pdf

### CODE REVIEW

**Review by bombasticSlacks — APPROVED** [review]
Looks good to me. I'm sleepy so I'll let at least 1 more set of eyes get on it

File: data/json/items/gun/45.json [hunk 1]
```diff
@@ -239,6 +239,43 @@
     "flags": [ "RELOAD_EJECT" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "45": 1 } } ]
   },
+  {
+    "id": "hk_usc45",
+    "copy-from": "rifle_semi",
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": { "str": "H&K USC 45 Carbine" },
+    "description": "A compact, light weight carbine chambering the popular 45 ACP pistol round.  Resembling the popular UMP submachine gun in both its blowback operating system and light weight polymer frame construction.  This civilian variant is distinguished by its skeletonized stock and 16 inch barrel.",
```
Comment (Night-Pryanik) [inline data/json/items/gun/45.json:248]:
```suggestion
    "description": "A compact, light weight carbine chambering the popular .45 ACP pistol round.  Resembling the popular UMP submachine gun in both its blowback operating system and light weight polymer frame construction.  This civilian variant is distinguished by its skeletonized stock and 16 inch barrel.",
```

File: data/json/items/gun/45.json [hunk 2]
```diff
@@ -239,6 +239,43 @@
...
+    "volume": "6116 ml",
+    "longest_side": "900 mm",
+    "price": 160000,
+    "price_postapoc": 3100,
+    "bashing": 12,
+    "material": [ "steel", "plastic" ],
+    "symbol": "(",
+    "color": "dark_gray",
+    "ammo": [ "45" ],
+    "ranged_damage": { "damage_type": "stab", "amount": 4 },
```
Comment (Night-Pryanik) [inline data/json/items/gun/45.json:259]:
```suggestion
    "ranged_damage": { "damage_type": "bullet", "amount": 4 },
```

**Comment by TheShadowFerret** [comment]
No bonus range like the CX4 Storm gives to 9mm?

**Comment by Ilurker** [comment]
> No bonus range like the CX4 Storm gives to 9mm?

That's a good point. Does anyone know the formula being used to determine range increases like that? Average carbine range seems to be approximately 36, it looks like the CX4 boosts the 9mm to 32. Applying the same +20 modifier would land the range of the USC at exactly 36 as well.

**Pull request merged**